inc/class-shortcake-bakery.php

```diff
@@ -118,4 +121,82 @@ public static function get_shortcake_admin_dependencies() {
 		return $r;
 	}
 
+	public function action_admin_enqueue_scripts() {
+		wp_enqueue_script( 'shortcake-bakery-admin', plugin_dir_url( dirname( __FILE__ ) ) . 'assets/js/build/shortcake-bakery-admin.js', array( 'media-views', 'shortcode-ui' ) );
+		wp_enqueue_style( 'shortcake-bakery', plugin_dir_url( dirname( __FILE__ ) ) . 'assets/css/shortcake-bakery.css' );
+
+		$strings = array(
+			'text' => array(
+				'addEmbed'           => esc_html__( 'Insert Embed Code', 'shortcake-bakery' ),
+				'insertButton'       => esc_html__( 'Insert embed', 'shortcake-bakery' ),
+				'customEmbedLabel'   => esc_html__( 'Paste any custom embed code here. If it matches a known post element, that post element will be used rather than the embed code.', 'shortcake-bakery' ),
+				'noReversalMatches'  => esc_html__( 'The embed code provided doesn\'t match any known post elements. This means that it may not display as expected.', 'shortcake-bakery' ),
+			),
+			'nonces' => array(
+				'customEmbedReverse' => wp_create_nonce( 'embed-reverse' ),
+			),
+		);
+
+		wp_localize_script( 'shortcake-bakery-admin', 'ShortcakeBakery', $strings );
+	}
+
+
+	/**
+	 * Output the "Add embed" button with WP media buttons.
+	 *
+	 * @return void
+	 */
+	public function action_media_buttons( $editor_id ) {
+		printf( '<button type="button" class="button insert-embed shortcake-bakery-insert-embed" data-editor="%s"><span class="dashicons dashicons-editor-code"></span> %s</button>',
+			esc_attr( $editor_id ),
+			esc_html__( 'Add Embed', 'fusion' )
+		);
+	}
+
+	public function action_ajax_shortcake_bakery_embed_reverse() {
+		if ( empty( $_POST['_wpnonce'] ) || empty( $_POST['custom_embed_code'] ) ) {
+			exit;
+		}
+
+		check_ajax_referer( 'embed-reverse', '_wpnonce' );
+		$provided_embed_code = wp_unslash( $_POST['custom_embed_code'] );
+		$result = $this->embed_reversal( $provided_embed_code );
+		do_action( 'shortcake_bakery_embed_reversal', $result, $provided_embed_code );
```

Comment:
Should we name this `shortcake_bakery_reversed_embed` to imply the action was completed?